modules/common/condition/types.go

```diff
@@ -41,17 +41,50 @@ const (
 )
 
 // Type - A summarizing name for a given condition
+// ---
+// Many .condition.type values are consistent across resources like Available, but because arbitrary conditions can be
+// useful (see .node.status.conditions), the ability to deconflict is important.
+// The regex it matches is (dns1123SubdomainFmt/)?(qualifiedNameFmt)
+// +kubebuilder:validation:Pattern=`^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])$`
+// +kubebuilder:validation:MaxLength=316
 type Type string
 
 // Reason - Why a particular condition is true, false or unknown
+// reason contains a programmatic identifier indicating the reason for the condition's last transition.
+// Producers of specific condition types may define expected values and meanings for this field,
+// and whether the values are considered a guaranteed API.
+// The value should be a CamelCase string.
+// This field may not be empty.
+// +kubebuilder:validation:MaxLength=1024
+// +kubebuilder:validation:MinLength=1
+// +kubebuilder:validation:Pattern=`^[A-Za-z]([A-Za-z0-9_,:]*[A-Za-z0-9_])?$`
 type Reason string
 
-// Condition defines an observation of a API resource operational state.
+// Condition contains details for one aspect of the current state of this API Resource.
+// ---
+// This struct is intended for direct use as an array at the field path .status.conditions.  For example,
+//
+//	type FooStatus struct{
+//	    // Represents the observations of a foo's current state.
+//	    // Known .status.conditions.type are: "Available", "Progressing", and "Degraded"
+//	    // +patchMergeKey=type
+//	    // +patchStrategy=merge
+//	    // +listType=map
+//	    // +listMapKey=type
+//	    Conditions []condition.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
+//
+//	    // other fields
+//	}
 type Condition struct {
 	// Type of condition in CamelCase.
+	// +required
+	// +kubebuilder:validation:Required
 	Type Type `json:"type"`
 
-	// Status of the condition, one of True, False, Unknown.
+	// status of the condition, one of True, False, Unknown.
+	// +required
```

Comment:
as Conditions got included into CRDs, we cannot change the optional field to become required, do we?
Also, how that change passes the crd checker? Let's try to include this into some operator as a dependency and see if crd checker allows that?